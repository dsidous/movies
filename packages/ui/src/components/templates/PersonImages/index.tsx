import React from 'react';
import Typography from '@material-ui/core/Typography';

import Section from '../../atoms/Section';
import MediaImage from '../../atoms/MediaImage';
import Spinner from '../../atoms/Spinner';
import PageTransition from '../../atoms/PageTransition';
import MiniHeader from '../../molecules/MiniHeader';
import useStyles from './styles';
import useConfig from '../../hooks/useConfig';
import { Person, Profile_Image } from '@typesRoots/person';

interface Props {
  loading: boolean;
  person: Person;
}

const PersonImages: React.FC<Props> = ({
  loading,
  person: { images, name, id, birthday, profile_path },
}) => {
  if (loading) {
    return <Spinner />;
  }

  const classes = useStyles();
  const { getImageURL } = useConfig();

  const list = ([] as Profile_Image[])
    .concat(images)
    .sort((a, b) => b.vote_average - a.vote_average)
    .map((image) => (
      <figure key={image.file_path} className={classes.item}>
        <a
          href={getImageURL({
            filePath: image.file_path,
            mediaType: 'profile',
            size: 3,
          })}
          target='_blank'
          rel='noopener noreferrer'
        >
          <MediaImage
            mediaType='profile'
            size={2}
            filePath={image.file_path}
            name={name}
          />
          <figcaption>
            <Typography variant='caption'>
              Size:
              <span>{` ${image.height} x ${image.width}`}</span>
            </Typography>
          </figcaption>
        </a>
      </figure>
    ));
  return (
    <PageTransition>
      <MiniHeader
        title={name}
        release_date={birthday}
        poster_path={profile_path}
        link={`/person/${id}`}
        linkCopy='Back to main'
      />
      <Section noborder disableGutters>
        <div className={classes.root}>{list}</div>
      </Section>
    </PageTransition>
  );
};

export default PersonImages;
