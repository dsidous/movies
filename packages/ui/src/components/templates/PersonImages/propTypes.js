import { object, bool } from 'prop-types';

export const propTypes = {
  loading: bool,
  person: object,
  config: object,
};

export const defaultProps = {};
