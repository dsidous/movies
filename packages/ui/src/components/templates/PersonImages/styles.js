import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(
  theme => ({
    root: {
      display: 'grid',
      gridGap: theme.spacing(2),
      gridTemplateColumns: 'repeat(2, 1fr)',

      [theme.breakpoints.up('sm')]: {
        gridGap: theme.spacing(5),
        gridTemplateColumns: 'repeat(4, 1fr)',
      },
    },

    item: {
      borderRadius: theme.shape.borderRadius,
      boxShadow: '0 0 20px 5px rgba(0, 0, 0, 0.1)',
      margin: 0,
      overflow: 'hidden',

      '& a': {
        color: '#000',

        '&:hover, &:visited, &:focus': {
          color: '#000',
          textDecoration: 'none',
        },
      },

      '& img': {
        width: '100%',
      },

      '& figcaption': {
        padding: theme.spacing(2),

        '& p': {
          margin: 0,
        },
      },
    },
  }),
  { name: 'personImages' },
);
