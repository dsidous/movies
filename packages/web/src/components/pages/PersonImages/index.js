import { compose, withProps } from 'recompose';
import { PersonImages } from '@movies/ui';
import { withPerson, withConfig } from '@movies/common';

export default compose(
  withProps(props => ({ personId: props.match.params.personId })),
  withPerson(),
  withConfig(),
)(PersonImages);
