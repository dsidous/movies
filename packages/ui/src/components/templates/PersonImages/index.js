/* eslint-disable camelcase */
import React from 'react';

import { propTypes } from './propTypes';
import MediaImage from '../../atoms/MediaImage';
import Spinner from '../../atoms/Spinner';
import PageTransition from '../../atoms/PageTransition';
import MiniHeader from '../../molecules/MiniHeader';

const PersonImages = ({
  loading,
  person: { images, name, id, birthday, profile_path },
  config: {
    images: { secure_base_url, profile_sizes },
  },
}) => {
  if (loading) {
    return <Spinner />;
  }

  const list = []
    .concat(images)
    .sort((a, b) => b.vote_average - a.vote_average)
    .map(image => (
      <figure key={image.file_path} className="cast-images__element">
        <a
          href={secure_base_url + profile_sizes[3] + image.file_path}
          target="_blank"
          rel="noopener noreferrer"
        >
          <MediaImage
            mediaType="profile"
            size={2}
            filePath={image.file_path}
            name={name}
          />
          <figcaption>
            <p>
              Size:
              <br />
              {`${image.height} x ${image.width}`}
            </p>
          </figcaption>
        </a>
      </figure>
    ));
  return (
    <PageTransition>
      <MiniHeader
        title={name}
        release_date={birthday}
        poster_path={profile_path}
        link={`/person/${id}`}
        linkCopy="Back to main"
      />

      <div className="cast-images__container">{list}</div>
    </PageTransition>
  );
};

PersonImages.propTypes = propTypes;

export default PersonImages;
