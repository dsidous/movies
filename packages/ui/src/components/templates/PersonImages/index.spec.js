import React from 'react';
import { shallow } from 'enzyme';
import PersonImages from '.';

const mockProps = {
  person: {
    images: [
      {
        file_path: 'path/to/image',
        vote_average: 1,
        height: 1,
        width: 1,
      },
      {
        file_path: 'path/to/image',
        vote_average: 2,
        height: 1,
        width: 1,
      },
    ],
    name: 'name',
    id: 1,
    birthday: '10.10.2010',
    profile_path: 'path/to/image',
  },
  config: {
    images: {
      secure_base_url: 'path/to/image',
      profile_sizes: ['1', '2', '3', '4'],
    },
  },
  loading: false,
};

describe('Templates/PersonImages', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<PersonImages {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render a spinner when loading expected', () => {
    const wrapper = shallow(<PersonImages {...mockProps} loading />);

    expect(wrapper).toMatchSnapshot();
  });
});
