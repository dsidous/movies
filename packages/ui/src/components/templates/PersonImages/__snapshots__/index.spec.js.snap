// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Templates/PersonImages should render a spinner when loading expected 1`] = `ShallowWrapper {}`;

exports[`Templates/PersonImages should render as expected 1`] = `ShallowWrapper {}`;
